add example for "Pulling a Backup with HTTP over a ssh tunnel"

JL710 opened the pull request:
What does this PR change? What problem does it solve?
-----------------------------------------------------

It adds a example in the docs for pulling a backup over a ssh tunnel as described [here](https://github.com/restic/restic/issues/299#issuecomment-3263824871).

Checklist
---------

- [X] I have added tests for all code changes.
- [X] I have added documentation for relevant changes (in the manual).
- [X] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [X] I'm done! This pull request is ready for review.

JL710 committed:
add example for "Pulling a Backup with HTTP over a ssh tunnel"

MichaelEischer commented on doc/080_examples.rst:
```diff
@@ -353,3 +353,56 @@ system.
+Run restic on the remote machine
+================================
+
+Then you can use restic through the ssh connection like this
```
Please describe the idea right at the start: the local host connects to the server and forwards rest-server to it. Then it runs restic on the remote server and sends the backup back to the local host via the tunnel.

After the backup is complete the ssh connection is closed again and the rest-server becomes inaccessible.

With that description in mind, I'd actually be amenable to use a title like the one you originally used `Pulling a Backup with rest-server over a ssh tunnel` .

MichaelEischer commented on doc/080_examples.rst:
```diff
@@ -353,3 +353,56 @@ system.
+
+.. code-block:: console
+
+   docker run --rm -e DISABLE_AUTHENTICATION=0 -p 2555:8000 -v /path/to/local/repo:/data --name restic_rest_server restic/rest-server
```
I think it would make sense to also pass `--append-only` to rest-server. Then the remote server also can't wipe the backup. It only has to be removed for `forget`/`prune` operations.

JL710 commented on doc/080_examples.rst:
```diff
@@ -353,3 +353,56 @@ system.
+   ssh -R 2555:localhost:2555 user@server_ip
+
+
+.. note:: ``-R 2555:localhost:2555`` (``local_port:localhost:remote_port``) specifies remote port forwarding → forwarding connections from the remote machine to the local machine
```
I am not sure what you mean.

JL710 committed:
apply suggestions from Michael Eischer

JL710 committed:
use rclone for rest server instead of docker

JL710 commented:
@codekow I did not know that. Using rclone is so much more convenient then docker. I added a commit that switches the docker part with rclone instead.

Not only is it more convenient, it also enables the user to use any file system that rclone can connect to without giving the backed up server access to it.

rawtaz committed:
doc: Reword parts of the text, replace rclone with rest-server

The intent here is to make the text more consistent in its use of different
concepts involved in explaining the idea and setup that is explained, and
to make it easier to follow.

We're also replacing rclone with rest-server, not because we dislike rclone
but in order to keep the text to the basic tooling and the main restic eco-
system.

Finally we also remove the previous tip at the end about keeping the SSH tunnel
up, as it will be during the time the SSH session is running (in which the user
is expected to run the restic commands).

MichaelEischer committed:
tweak rest-server call and restore rclone as alternative

MichaelEischer reviewed (approved):
LGTM. I've added a reworked version by @rawtaz to the PR. The prime tool is now rest-server, but rclone is also described as an alternative.
